src/shogun/lib/DynamicArray.h

```diff
@@ -13,30 +13,875 @@
 
 #include <shogun/lib/config.h>
 
-#include <shogun/base/SGObject.h>
 #include <shogun/base/DynArray.h>
 #include <shogun/base/Parameter.h>
+#include <shogun/base/SGObject.h>
+#include <vector>
+
+namespace shogun
+{
+	/** @brief Template Dynamic array class that creates an array that can
+	 * be used like a list or an array.
+	 *
+	 * It grows and shrinks dynamically, while elements can be accessed
+	 * via index.  It is performance tuned for simple types like float
+	 * etc. and for hi-level objects only stores pointers, which are not
+	 * automagically SG_REF'd/deleted.
+	 */
+	template <>
```

Comment:
this is true for the real template definition as well, i.e. it should be:
```template <class T> class CDynamicArray : public CSGObject```

instead of
```
       template <class T>
       class CDynamicArray : public CSGObject
```
this will fix the problem of `class_list.cpp`